homeassistant/components/august/gateway.py

```diff
@@ -0,0 +1,144 @@
+"""Handle August connection setup and authentication."""
+
+import asyncio
+import logging
+
+from august.api import Api
+from august.authenticator import AuthenticationState, Authenticator
+from requests import RequestException, Session
+
+from homeassistant.const import CONF_PASSWORD, CONF_TIMEOUT, CONF_USERNAME
+
+from .const import (
+    CONF_ACCESS_TOKEN_CACHE_FILE,
+    CONF_INSTALL_ID,
+    CONF_LOGIN_METHOD,
+    DEFAULT_AUGUST_CONFIG_FILE,
+    VERIFICATION_CODE_KEY,
+)
+from .exceptions import CannotConnect, InvalidAuth, RequireValidation
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AugustGateway:
+    """Handle the connection to August."""
+
+    def __init__(self, hass):
+        """Init the connection."""
+        try:
+            self._api_http_session = Session()
+        except RequestException as ex:
+            _LOGGER.warning("Creating HTTP session failed with: %s", str(ex))
+        self._token_refresh_lock = asyncio.Lock()
+        self._hass = hass
+        self._config = None
+        self._api = None
+        self._authenticator = None
+        self._authentication = None
+
+    @property
+    def authenticator(self):
+        """August authentication object from py-august."""
+        return self._authenticator
+
+    @property
+    def authentication(self):
+        """August authentication object from py-august."""
+        return self._authentication
+
+    @property
+    def access_token(self):
+        """Access token for the api."""
+        return self._authentication.access_token
+
+    @property
+    def api(self):
+        """August api object from py-august."""
+        return self._api
+
+    def config_entry(self):
+        """Config entry."""
+        return {
+            CONF_LOGIN_METHOD: self._config[CONF_LOGIN_METHOD],
+            CONF_USERNAME: self._config[CONF_USERNAME],
+            CONF_PASSWORD: self._config[CONF_PASSWORD],
+            CONF_INSTALL_ID: self._config.get(CONF_INSTALL_ID),
+            CONF_TIMEOUT: self._config.get(CONF_TIMEOUT),
+            CONF_ACCESS_TOKEN_CACHE_FILE: self._config[CONF_ACCESS_TOKEN_CACHE_FILE],
+        }
+
+    def setup(self, conf):
+        """Create the api and authenticator objects."""
+        if not conf.get(VERIFICATION_CODE_KEY):
```

Comment:
Use a guard clause to save indentation.

```suggestion
        if not conf.get(VERIFICATION_CODE_KEY):
            return
```

Reply:
done